sematic/ui/packages/main/src/tests/test_utils.cy.ts

```diff
@@ -15,4 +15,48 @@ describe('abbreviatedUserName()', () => {
     it("should return empty string when the parameter is not provided", () => {
         expect(abbreviatedUserName(null)).to.equal("");
     });
+});
+
+describe('AsyncInvocationQueue', () => {
+    it("should sequentialize function invocation using a queue", () => {
+        let asyncFunction: any;
+        let acquire: any;
+
+        cy.clock();
+        cy.wait(0).then(async () => {
+            acquire = AsyncInvocationQueue();
+            asyncFunction = cy.spy(async () => {
+                await new Promise(resolve => setTimeout(resolve, 100));
+            });
+        });
+        cy.wait(0).then(async () => {
+            setTimeout(async () => {
+                const release = await acquire();
+                await asyncFunction();
+                release();
+            }, 0);
+    
+            setTimeout(async () => {
+                const release = await acquire();
+                await asyncFunction();
+                release();
+            }, 20);
+        });
+
+        cy.tick(0); // trigger the first call
+        cy.tick(20); // trigger the second call
```

Comment:
Are these UI tests executing in parallel mode? I want to avoid adding more tests like these in the future impacting the execution time.

Can these numbers be reduced?

Reply:
These use a simulated clock; the point is to make sure they don't elapse that much actual time, but quickly dial the clock to advance to the next moment. All the JS facilities like `setTimeout` will move with the simulated clock. So we will save time on tests.

The numbers are specially picked to work with the test case and the internal mechanisms of `AsyncInvocationQueue.` 

Given that, are they still concerns?

Reply:
Nope, that's great! Thank you!